src/Cassandra/Data/Linq/CqlExpressionVisitor.cs

```diff
@@ -867,19 +892,33 @@ protected override Expression VisitBinary(BinaryExpression node)
                             return node;
                         }
                     }
+                    else if (node.NodeType == ExpressionType.Equal && IsBoolMember(node.Left))
+                    {
+                        // Handle x.prop == boolValue explicitly
+                        FillBooleanCondition((MemberExpression)node.Left, clause);
+                        Visit(node.Right);
+                        return node;
+                    }
+                    else if (node.NodeType == ExpressionType.Equal && IsBoolMember(node.Right))
+                    {
+                        // Handle boolValue == x.prop explicitly
+                        FillBooleanCondition((MemberExpression)node.Right, clause);
+                        Visit(node.Left);
+                        return node;
+                    }
                     else
                     {
                         Visit(DropNullableConversion(node.Left));
-                        _currentCondition.Get().Item1.Append(" " + CqlTags[node.NodeType] + " ");
+                        clause.Append(" " + CqlTags[node.NodeType] + " ");
                         Visit(DropNullableConversion(node.Right));
                         return node;
                     }
                 }
                 else if (!CqlUnsupTags.Contains(node.NodeType))
                 {
                     var val = Expression.Lambda(node).Compile().DynamicInvoke();
-                    _currentCondition.Get().Item2.Add(val);
-                    _currentCondition.Get().Item1.Append("?");
+                    parameters.Add(val);
+                    clause.Append("?");
```

Comment:
There are some occurrences of the string hardcoded "?", and as it carry a strong meaning (parameter value) I would refactor to create a constant.